File: beLibrary/src/main/java/com/example/btl/entity/CallCard.java
package com.example.btl.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class CallCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String borrowDate;
    private String dueDate;
    private String state;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;
    private String note;


}

File: LibraryManagement/app/src/main/java/com/example/librarymanagement/models/CallCard.kt
package com.example.librarymanagement.models

import java.io.Serializable

data class CallCard(
    val id: Int? = null,
    var books: List<BookDetailResponse>? = null,
    val user: User? = null,
    var borrowDate: String? = null,
    var dueDate: String? = null,
    var state: String? = null,
    var note: String? = null,
):Serializable{
}

class CallCardRequest {
     var id: Long? = null
     var borrowDate: String? = null
     var dueDate: String? = null
     var state: String? = null
    var note: String? = null
     var userId: Long? = null
     var books: List<BookDetailResponse>? = null
}

data class HistoryCallCard(
    var books: BookDetailResponse? = null,
    var borrowDate: String? = null,
    var dueDate: String? = null,
    var state: String? = null,
    var note: String? = null,
):Serializable{
}

data class Statistic(
    var user: User? = null,
    var borrowDate: String? = null,
    var dueDate: String? = null,
    var state: String? = null,
    var note: String? = null,
):Serializable{
}